shared/fs/filepreview/view-container.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+import {connect, type Dispatch, type TypedState} from '../../util/container'
+import * as Constants from '../../constants/fs'
+import * as FsGen from '../../actions/fs-gen'
+import * as React from 'react'
+import * as Types from '../../constants/types/fs'
+import DefaultView from './default-view-container'
+import ImageView from './image-view'
+import TextView from './text-view'
+import {Text} from '../../common-adapters'
+
+type Props = {
+  path: Types.Path,
+  fileViewType?: Types.FileViewType, // can be set by default-view-container.js for type override
+}
+
+const mapStateToProps = (state: TypedState) => ({
+  _serverInfo: state.fs.localHTTPServerInfo,
+})
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  onInvalidToken: () => dispatch(FsGen.createRefreshLocalHTTPServerInfo()),
```

Comment:
Do we want to trigger this action on load too?

Reply:
We do trigger this in `actions/notifications.js` along with a bunch of other initializations, i.e. when app starts. The token doesn't get invalidated unless it gets pushed out of cache, or `kbfsfuse` crashes. So this really happens very rarely.

Reply:
cool thx